### ISSUE TITLE
Training coco metrics seems to be good but when trained model validated using --eval mode, only garbage values are printed.

### ISSUE THREAD

**Issue by purvang3** [gaopengcuhk/SMCA-DETR#21]
Thank you for the amazing work. 
I have trained model for 50 epochs and while training, periodic eval seems to be reasonable and improving.

![image](https://user-images.githubusercontent.com/39809304/158005413-6e3ce357-3e53-4659-b090-4e8061d9e399.png)

But then, when I use trained model on same validation data by using --eval argument and defining path for "checkpoint.pth" in d2/configs/detr_256_6_6_torchvision.yaml config, I am getting all zeros.

my ground truth val.json in (xywh) format.
After investigating more, 

I found that DataLoader loads ground truth boxes in normalize form but predicted output boxes, those are in unnormalized form 
because of following lines from engine.py in evaluate function.

"
orig_target_sizes = torch.stack([t["orig_size"] for t in targets], dim=0)
results = postprocessors['bbox'](outputs, orig_target_sizes)
"

even after fixing predicted bbox to normalize or unnormalize groud truth bbox in DataLoader class, still I am getting all zeros in coco detection metrics.

I am not sure though training script call same function for periodic evaluation, why using --eval argument on same data produces completely different result?

Am I using some different model instead of trained one? I changed weight field in d2/configs/detr_256_6_6_torchvision.yaml config to model's "checkpoint.pth" .

Please let me know if you need any additional information.
Thanks.

**Comment by purvang3** [comment]
@gaopengcuhk Could you please point to the document or mention steps on how to evaluate trained model? Thank you

**Comment by hanqiu-hq** [comment]
We did not implement the SMCA based on detectron2(d2). You can use the following command to evaluate the trained model:
```
python -m torch.distributed.launch --nproc_per_node=8 --use_env main.py --coco_path /path/to/coco --batch_size 2 --lr_drop 40 --num_queries 300 --epochs 50 --dynamic_scale type3 --output_dir smca_single_scale --eval --resume /path/to/checkpoint.pth
```

**Comment by purvang3** [comment]
Thank you @gaopengcuhk
